src/model.js

```diff
@@ -175,9 +175,21 @@ var Model = TransisObject.extend(function() {
 
     if (typeof f === 'function') { f.call(subclass); }
 
+    if(this.displayName !== 'Transis.Model') {
+      this.subclasses = this.subclasses || [];
+      this.subclasses.push(name);
```

Comment:
Actually, this would be necessary in order to ensure that you are getting an actual subclass of the receiver.